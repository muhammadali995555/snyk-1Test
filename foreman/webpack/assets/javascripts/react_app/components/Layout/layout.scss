@import '~@theforeman/vendor/scss/variables';
@import '../../common/colors.scss';
@import '../../common/variables.scss';

.secondary-nav-item-pf > .nav-pf-secondary-nav {
  background-color: $nav-pf-vertical-active-bg-color;

  .mobile-active {
    // Mark active Location/Org in Mobile view
    a {
      .list-group-item-value {
        background-color: $nav-pf-vertical-secondary-active-bg-color;
      }
    }
  }

  .list-group-item {
    a {
      background-color: $nav-pf-vertical-active-bg-color;

      &:hover {
        .list-group-item-value {
          background-color: $nav-pf-vertical-secondary-active-bg-color;
        }
      }
    }

    span {
      color: $nav-pf-vertical-active-icon-color;
    }
  }
}

#react-app-root {
  .react-container {
    margin-right: auto;
    margin-left: 200px;
    padding-left: 20px;
    padding-right: 20px;
  }

  .nav-pf-vertical {
    background-color: $nav-pf-vertical-secondary-active-bg-color;

    .list-group {
      .list-group-item {
        a {
          span {
            color: $nav-pf-vertical-icon-color;
          }

          &:hover {
            background-color: $nav-pf-vertical-active-bg-color;

            span {
              color: $nav-pf-vertical-active-icon-color;
            }
          }

          &::after {
            color: $nav-pf-vertical-icon-color;
          }
        }

        &:hover {
          a {
            background-color: $nav-pf-vertical-active-bg-color;

            span {
              color: $nav-pf-vertical-active-icon-color;
            }
          }
        }

        &.active {
          a {
            background-color: $nav-pf-vertical-active-bg-color;

            span {
              color: $nav-pf-vertical-active-icon-color;
            }
          }
        }
      }
    }
  }

  .secondary-nav-item-pf.hidden-nav-lg {
    display: none;

    @media (max-width: $header-max-width) {
      display: block;
    }
  }
}

/*
  patternfly-sass hides the vertical-nav on smaller screens, see: https://github.com/patternfly/patternfly-sass/blob/5e7cce3445d5b1af1e16500695b9f33edc6a4f6c/assets/javascripts/patternfly-functions.js#L260
  the react-container needs to be shifted when it happens.
*/
@media only screen and (max-width: $pf-global--breakpoint--md) {
  #react-app-root .react-container,
  #rails-app-content {
    margin-left: 0;
  }
}

#account_menu {
  outline: none;
}

#vertical-spinner {
  margin-top: 15px;
  margin-left: 5px;
}

.navbar-header {
  width: 42px;
}

#navbar-header {
  left: 42px;
}

#data-toolbar {
  padding: 0;
  background-color: transparent;
}

.pf-c-page {
  background-color: rgba(0, 0, 0, 0);
}

.collapsed-nav {
  #react-app-root .react-container,
  #rails-app-content {
    margin-left: 75px;
  }

  @media only screen and (max-width: $pf-global--breakpoint--md) {
    #react-app-root .react-container,
    #rails-app-content {
      margin-left: 0;
    }
  }
}
