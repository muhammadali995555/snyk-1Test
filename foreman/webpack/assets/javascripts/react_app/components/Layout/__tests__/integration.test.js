import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { IntegrationTestHelper } from '@theforeman/test';

import { hasTaxonomiesMock } from '../Layout.fixtures';
import Layout, { reducers } from '../index';
import ForemanContext from '../../../Root/Context/ForemanContext';

jest.mock('../../notifications', () => 'span');

// mock the taxonomy context to match the test until we implement context setter

jest
  .spyOn(ForemanContext, 'useForemanLocation')
  .mockReturnValue({ title: 'london' });
jest
  .spyOn(ForemanContext, 'useForemanOrganization')
  .mockReturnValue({ title: 'org1' });

afterEach(() => {
  jest.clearAllMocks();
});
describe('Layout integration test', () => {
  it('should flow', async () => {
    const integrationTestHelper = new IntegrationTestHelper(reducers);
    Object.defineProperty(window, 'location', {
      writable: true,
      value: { assign: jest.fn(), pathname: '/' },
    });
    const component = integrationTestHelper.mount(
      <Router>
        <Layout {...hasTaxonomiesMock} />
      </Router>
    );
    await IntegrationTestHelper.flushAllPromises();
    component.update();

    integrationTestHelper.takeStoreSnapshot('initial state');
    expect(
      component
        .find('#location-dropdown .pf-c-context-selector__toggle-text')
        .text()
    ).toBe('london');
    expect(
      component
        .find('#organization-dropdown .pf-c-context-selector__toggle-text')
        .text()
    ).toBe('org1');

    const hostsMenuItem = component.find('.secondary-nav-item-pf > a');

    hostsMenuItem.at(1).simulate('click');
    integrationTestHelper.takeStoreAndLastActionSnapshot(
      'Changed ActiveMenu to Hosts'
    );
    expect(component.find('.secondary-nav-item-pf .active > a').text()).toBe(
      'Hosts'
    );
  });
});
