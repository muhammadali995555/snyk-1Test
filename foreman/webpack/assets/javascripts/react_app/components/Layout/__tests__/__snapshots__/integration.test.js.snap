// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout integration test should flow: Changed ActiveMenu to Hosts 1`] = `
Object {
  "action": Array [
    Array [
      Object {
        "payload": Object {
          "activeMenu": "Hosts",
        },
        "type": "LAYOUT_CHANGE_ACTIVE",
      },
    ],
  ],
  "state": Object {
    "layout": Object {
      "activeMenu": "Hosts",
      "currentLocation": "london",
      "currentOrganization": "org1",
      "isCollapsed": false,
      "isLoading": false,
      "items": Array [
        Object {
          "children": Array [
            Object {
              "exact": true,
              "name": "Dashboard",
              "type": "item",
              "url": "/",
            },
            Object {
              "name": "Facts",
              "type": "item",
              "url": "/fact_values",
            },
          ],
          "className": "",
          "icon": "fa fa-tachometer",
          "name": "Monitor",
          "type": "sub_menu",
        },
        Object {
          "children": Array [
            Object {
              "name": "All Hosts",
              "type": "item",
              "url": "/hosts/new",
            },
          ],
          "className": "",
          "icon": "fa fa-server",
          "name": "Hosts",
          "type": "sub_menu",
        },
        Object {
          "children": Array [
            Object {
              "name": undefined,
              "type": "item",
              "url": "/nameless",
            },
            Object {
              "name": undefined,
              "type": "divider",
            },
          ],
          "className": "",
          "icon": "pficon pficon-unplugged",
          "name": "Empty",
          "type": "sub_menu",
        },
        Object {
          "children": Array [
            Object {
              "name": "Any Organization",
              "onClick": [Function],
            },
            Object {
              "name": "org1",
              "onClick": [Function],
              "type": undefined,
            },
            Object {
              "name": "org2",
              "onClick": [Function],
              "type": undefined,
            },
          ],
          "className": "organization-menu hidden-nav-lg",
          "icon": "fa fa-building",
          "name": "Organizations",
          "type": "sub_menu",
        },
        Object {
          "children": Array [
            Object {
              "name": "Any Location",
              "onClick": [Function],
            },
            Object {
              "name": "yaml",
              "onClick": [Function],
              "type": undefined,
            },
            Object {
              "name": "london",
              "onClick": [Function],
              "type": undefined,
            },
            Object {
              "name": "norway",
              "onClick": [Function],
              "type": undefined,
            },
          ],
          "className": "location-menu hidden-nav-lg",
          "icon": "fa fa-globe",
          "name": "Locations",
          "type": "sub_menu",
        },
      ],
    },
  },
}
`;

exports[`Layout integration test should flow: initial state 1`] = `
Object {
  "layout": Object {
    "activeMenu": "Monitor",
    "currentLocation": "london",
    "currentOrganization": "org1",
    "isCollapsed": false,
    "isLoading": false,
    "items": Array [
      Object {
        "children": Array [
          Object {
            "exact": true,
            "name": "Dashboard",
            "type": "item",
            "url": "/",
          },
          Object {
            "name": "Facts",
            "type": "item",
            "url": "/fact_values",
          },
        ],
        "className": "",
        "icon": "fa fa-tachometer",
        "name": "Monitor",
        "type": "sub_menu",
      },
      Object {
        "children": Array [
          Object {
            "name": "All Hosts",
            "type": "item",
            "url": "/hosts/new",
          },
        ],
        "className": "",
        "icon": "fa fa-server",
        "name": "Hosts",
        "type": "sub_menu",
      },
      Object {
        "children": Array [
          Object {
            "name": undefined,
            "type": "item",
            "url": "/nameless",
          },
          Object {
            "name": undefined,
            "type": "divider",
          },
        ],
        "className": "",
        "icon": "pficon pficon-unplugged",
        "name": "Empty",
        "type": "sub_menu",
      },
      Object {
        "children": Array [
          Object {
            "name": "Any Organization",
            "onClick": [Function],
          },
          Object {
            "name": "org1",
            "onClick": [Function],
            "type": undefined,
          },
          Object {
            "name": "org2",
            "onClick": [Function],
            "type": undefined,
          },
        ],
        "className": "organization-menu hidden-nav-lg",
        "icon": "fa fa-building",
        "name": "Organizations",
        "type": "sub_menu",
      },
      Object {
        "children": Array [
          Object {
            "name": "Any Location",
            "onClick": [Function],
          },
          Object {
            "name": "yaml",
            "onClick": [Function],
            "type": undefined,
          },
          Object {
            "name": "london",
            "onClick": [Function],
            "type": undefined,
          },
          Object {
            "name": "norway",
            "onClick": [Function],
            "type": undefined,
          },
        ],
        "className": "location-menu hidden-nav-lg",
        "icon": "fa fa-globe",
        "name": "Locations",
        "type": "sub_menu",
      },
    ],
  },
}
`;
